File: Experiment06_Matrix_Reflection.m
clc; clear; close all;
P = [1 3 2 1; 1 1 3 1];
R = [1 0; 0 -1];
P_new = R * P;
plot(P(1,:), P(2,:), 'b', 'LineWidth', 2); hold on;
plot(P_new(1,:), P_new(2,:), 'r', 'LineWidth', 2);
legend('Original', 'Reflected');
title('Matrix Reflection');
grid on; axis equal;

File: Experiment06_Matrix_Reflection.py
import numpy as np
import matplotlib.pyplot as plt
P = np.array([[1,3,2,1],[1,1,3,1]])
R = np.array([[1,0],[0,-1]])
P_new = R @ P
plt.plot(P[0],P[1])
plt.plot(P_new[0],P_new[1])
plt.title("Matrix Reflection")
plt.gca().set_aspect('equal')
plt.show()